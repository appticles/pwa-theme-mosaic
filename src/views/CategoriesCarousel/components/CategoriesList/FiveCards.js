import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const FiveCards = props => (
  <Grid columns={2} divided className="one-third-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} key={props.categoriesList[2].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} key={props.categoriesList[3].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[4]} key={props.categoriesList[4].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FiveCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default FiveCards;
