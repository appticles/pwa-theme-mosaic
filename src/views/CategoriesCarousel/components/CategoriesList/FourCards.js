import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const FourCards = props => (
  <Grid columns={2} divided className="half-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} key={props.categoriesList[2].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} key={props.categoriesList[3].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FourCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default FourCards;
