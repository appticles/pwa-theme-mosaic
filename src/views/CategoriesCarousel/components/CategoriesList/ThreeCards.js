import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const ThreeCards = props => (
  <Grid columns={2} divided className="half-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} key={props.categoriesList[2].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

ThreeCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default ThreeCards;
