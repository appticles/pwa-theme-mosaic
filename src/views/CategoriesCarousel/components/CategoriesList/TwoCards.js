import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const TwoCards = props => (
  <Grid columns={1} divided className="half-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

TwoCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default TwoCards;
