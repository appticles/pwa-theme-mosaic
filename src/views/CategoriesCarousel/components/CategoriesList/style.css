/* override semantic ui styles */
/* GRID */
.categories-card .ui.grid {
  margin: 0; }
  .categories-card .ui.grid > .row {
    justify-content: center;
    padding-top: 0;
    padding-bottom: 0; }
    .categories-card .ui.grid > .row > .column {
      padding-left: 0;
      padding-right: 0; }

.categories-card .ui.divided.grid:not([class*="vertically divided"]) > .row > .column {
  box-shadow: none; }

.categories-card .divided.two > .row > .column:only-child {
  width: 100vw; }

.categories-card .full-row .row {
  height: 100%; }

.categories-card .half-row .row {
  height: 50.2%; }

.categories-card .one-third-row .row {
  height: 33.7%; }
