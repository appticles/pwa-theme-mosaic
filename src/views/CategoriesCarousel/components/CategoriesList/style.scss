/* override semantic ui styles */
/* GRID */

.categories-card {
  .ui.grid {
    margin: 0;
    > .row {
      justify-content: center;
      padding-top: 0;
      padding-bottom: 0;
      > .column {
        padding-left: 0;
        padding-right: 0;
      }
    }
  }
  .ui.divided.grid:not([class*="vertically divided"]) > .row > .column {
    box-shadow: none;
  }
  .divided.two > .row > .column:only-child {
    width: 100vw;
  }
  .full-row .row {
    height: 100%;
  }
  .half-row .row {
    height: 50.2%;
  }
  .one-third-row .row {
    height: 33.7%;
  }
}
