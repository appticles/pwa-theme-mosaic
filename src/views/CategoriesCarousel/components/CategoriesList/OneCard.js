import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';
import './style.css';

const OneCard = props => (
  <Grid className="full-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

OneCard.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};
export default OneCard;
