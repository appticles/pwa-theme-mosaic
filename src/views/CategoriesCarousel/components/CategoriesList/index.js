import React from 'react';
import PropTypes from 'prop-types';

import { categoryPropType } from '../../reducer';
import FiveCards from './FiveCards';
import FourCards from './FourCards';
import ThreeCards from './ThreeCards';
import TwoCards from './TwoCards';
import OneCard from './OneCard';

import './style.css';

const CategoriesList = (props) => {
  function getCards() {
    switch (props.categoriesChunk.length % 5) {
      case 0:
        return <FiveCards categoriesList={props.categoriesChunk} />;
      case 1:
        return <OneCard categoriesList={props.categoriesChunk} />;
      case 2:
        return <TwoCards categoriesList={props.categoriesChunk} />;
      case 3:
        return <ThreeCards categoriesList={props.categoriesChunk} />;
      case 4:
        return <FourCards categoriesList={props.categoriesChunk} />;
      default:
        return null;
    }
  }

  return <div>{props.categoriesChunk.length !== 0 ? getCards() : ''}</div>;
};

CategoriesList.propTypes = {
  categoriesChunk: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default CategoriesList;
