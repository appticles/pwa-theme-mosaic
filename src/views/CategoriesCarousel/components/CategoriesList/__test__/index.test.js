import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import CategoriesList from '../index';

describe('Categories list', () => {
  it('should render correctly 5 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
      {
        slug: 'mock-slug-2',
        id: 2,
        name: 'Some other mock name',
        link: 'http://localhost/mock-slug-2',
      },
      {
        slug: 'mock-slug-3',
        id: 3,
        name: 'Some name',
        link: 'http://localhost/mock-slug-3',
      },
      {
        slug: 'mock-slug-4',
        id: 4,
        name: 'Mock name',
        link: 'http://localhost/mock-slug-4',
      },
      {
        slug: 'mock-slug-5',
        id: 5,
        name: 'Some other other mock name',
        link: 'http://localhost/mock-slug-5',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 4 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
      {
        slug: 'mock-slug-2',
        id: 2,
        name: 'Some other mock name',
        link: 'http://localhost/mock-slug-2',
      },
      {
        slug: 'mock-slug-3',
        id: 3,
        name: 'Some name',
        link: 'http://localhost/mock-slug-3',
      },
      {
        slug: 'mock-slug-4',
        id: 4,
        name: 'Mock name',
        link: 'http://localhost/mock-slug-4',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 3 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
      {
        slug: 'mock-slug-2',
        id: 2,
        name: 'Some other mock name',
        link: 'http://localhost/mock-slug-2',
      },
      {
        slug: 'mock-slug-3',
        id: 3,
        name: 'Some name',
        link: 'http://localhost/mock-slug-3',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly 2 categories', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
      {
        slug: 'mock-slug-2',
        id: 2,
        name: 'Some other mock name',
        link: 'http://localhost/mock-slug-2',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly one category', () => {
    const mockCategoriesList = [
      {
        slug: 'mock-slug-1',
        id: 1,
        name: 'Some mock name',
        link: 'http://localhost/mock-slug-1',
      },
    ];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render correctly no category', () => {
    const mockCategoriesList = [];

    const output = shallow(<CategoriesList categoriesChunk={mockCategoriesList} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
