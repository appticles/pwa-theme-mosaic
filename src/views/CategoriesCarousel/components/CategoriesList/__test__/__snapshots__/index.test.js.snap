// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories list should render correctly 2 categories 1`] = `
<div>
  <TwoCards
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 2,
          "name": "Some other mock name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;

exports[`Categories list should render correctly 3 categories 1`] = `
<div>
  <ThreeCards
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 2,
          "name": "Some other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 3,
          "name": "Some name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;

exports[`Categories list should render correctly 4 categories 1`] = `
<div>
  <FourCards
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 2,
          "name": "Some other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 3,
          "name": "Some name",
          "slug": "mock-slug",
        },
        Object {
          "id": 4,
          "name": "Mock name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;

exports[`Categories list should render correctly 5 categories 1`] = `
<div>
  <FiveCards
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 2,
          "name": "Some other mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 3,
          "name": "Some name",
          "slug": "mock-slug",
        },
        Object {
          "id": 4,
          "name": "Mock name",
          "slug": "mock-slug",
        },
        Object {
          "id": 5,
          "name": "Some other other mock name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;

exports[`Categories list should render correctly no category 1`] = `<div />`;

exports[`Categories list should render correctly one category 1`] = `
<div>
  <OneCard
    categoriesList={
      Array [
        Object {
          "id": 1,
          "name": "Some mock name",
          "slug": "mock-slug",
        },
      ]
    }
  />
</div>
`;
